clients/src/main/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImpl.java

```diff
@@ -741,61 +767,86 @@ boolean reconcile() {
                 revokedPartitions
         );
 
-        CompletableFuture<Void> revocationResult;
-        if (!revokedPartitions.isEmpty()) {
-            revocationResult = revokePartitions(revokedPartitions);
+        // Commit offsets if auto-commit enabled before reconciling a new assignment. Request will
+        // be retried until it succeeds, fails with non-retriable error, or timer expires.
+        CompletableFuture<Void> commitResult;
+
+        if (commitRequestManager.autoCommitEnabled()) {
```

Comment:
`autoCommitAllConsumedNow` checks autoCommitEnabled internally, so we should be able to eliminate this if